pytorch_lightning/metrics/classification/utils.py

```diff
@@ -0,0 +1,382 @@
+    not check for consistency of number of classes, other functions take
+    care of that.
+
+    It returns the name of the case in which the inputs fall, and the implied
+    number of classes (from the C dim for multi-class data, or extra dim(s) for
+    multi-label data).
+    """
+
+    preds_float = preds.is_floating_point()
+
+    if preds.ndim == target.ndim:
+        if preds.shape != target.shape:
+            raise ValueError(
+                "The `preds` and `target` should have the same shape,",
+                f" got `preds` shape = {preds.shape} and `target` shape = {target.shape}.",
+            )
+        if preds_float and target.max() > 1:
+            raise ValueError(
+                "If `preds` and `target` are of shape (N, ...) and `preds` are floats, `target` should be binary."
+            )
+
+        # Get the case
+        if preds.ndim == 1 and preds_float:
+            case = "binary"
+        elif preds.ndim == 1 and not preds_float:
+            case = "multi-class"
+        elif preds.ndim > 1 and preds_float:
+            case = "multi-label"
+        else:
+            case = "multi-dim multi-class"
+
+        implied_classes = torch.prod(torch.Tensor(list(preds.shape[1:])))
+
+    elif preds.ndim == target.ndim + 1:
+        if not preds_float:
+            raise ValueError("If `preds` have one dimension more than `target`, `preds` should be a float tensor.")
+        if preds.shape[2:] != target.shape[1:]:
+            raise ValueError(
+                "If `preds` have one dimension more than `target`, the shape of `preds` should be"
+                " (N, C, ...), and the shape of `target` should be (N, ...)."
+            )
+
+        implied_classes = preds.shape[1]
+
+        if preds.ndim == 2:
+            case = "multi-class"
+        else:
+            case = "multi-dim multi-class"
+    else:
+        raise ValueError(
+            "The `preds` and `target` should both have the (same) shape (N, ...), or `target` (N, ...)"
+            " and `preds` (N, C, ...)."
+        )
+
+    return case, implied_classes
+
+
+def _check_num_classes_binary(num_classes: int, is_multiclass: bool):
+    """
+    This checks that the consistency of `num_classes` with the data
+    and `is_multiclass` param for binary data.
+    """
+
+    if num_classes > 2:
+        raise ValueError("Your data is binary, but `num_classes` is larger than 2.")
+    elif num_classes == 2 and not is_multiclass:
+        raise ValueError(
+            "Your data is binary and `num_classes=2`, but `is_multiclass` is not True."
+            " Set it to True if you want to transform binary data to multi-class format."
+        )
+    elif num_classes == 1 and is_multiclass:
+        raise ValueError(
+            "You have binary data and have set `is_multiclass=True`, but `num_classes` is 1."
+            " Either leave `is_multiclass` unset or set it to 2 to transform binary data to multi-class format."
```

Comment:
By default `is_multiclass=None`. I see there's some misunderstanding about the use of this parameter - I'll write a longer explanation here (since it seems to be the only issue remaining) and improve the description in the docs.

Reply:
updated.
BTW you can go to `Files Changed` section and add multiple suggestions in a single commit

Reply:
Thanks, didn't know that :)